File: custom_view/MyFlipView/src/main/java/com/example/myflipview/MainActivity.java
package com.example.myflipview;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        FlipView flipView = findViewById(R.id.flip_view);
        flipView.setScrollForwardDirection(false);
        flipView.flipTo(2);

        View button = findViewById(R.id.button);
        button.setOnClickListener(v -> {
            flipView.flipTo(5);
        });
    }
}

File: study_abroad/HenCoder/MaterialEditText/src/main/java/com/xuebinduan/materialedittext/MainActivity.kt
package com.xuebinduan.materialedittext

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.view.MotionEvent
import androidx.core.view.postDelayed

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val met = findViewById<MaterialEditText>(R.id.met)
        met.postDelayed(3000){
            met.useFloatingLabel = false
        }

        Handler().post {

        }

    }
}